org.mwc.cmap.TimeController/src/org/mwc/cmap/TimeController/recorders/CoordinateRecorder.java

```diff
@@ -144,14 +149,297 @@ protected void showMessageDialog(final String message)
   {
     Display.getDefault().asyncExec(new Runnable()
     {
-      
+
       @Override
       public void run()
       {
         MessageDialog.open(MessageDialog.INFORMATION, Display.getDefault()
-            .getActiveShell(), "Export", message, MessageDialog.INFORMATION);    
+            .getActiveShell(), "Export", message, MessageDialog.INFORMATION);
       }
     });
-    
+
+  }
+
+  public static class CoordinateRecorderTest extends junit.framework.TestCase
+  {
+    static public final String TEST_ALL_TEST_TYPE = "UNIT";
+
+    static final String collingwood = "COLLINGWOOD";
+    static final String nelson = "NELSON";
+
+    public CoordinateRecorder getRecorder()
+    {
+      final File testFile = new File(
+          "../org.mwc.cmap.combined.feature/root_installs/sample_data/offset_times.rep");
+
+      // ok, now try to read it in
+      final Layers _theLayers = new Layers();
+
+      // add the REP importer
+      MWC.Utilities.ReaderWriter.ImportManager.addImporter(

```

Comment:
I _think_ we can start off further down the call chain here.  I _think_ we can call ImportReplay directly to load the data into the `Layers` object.  That would avoid the thread-related code that is necessary in the `BaseImportCaller` implementation.